platform/platform-paxweb-jettyconfig/src/main/java/org/codice/ddf/security/session/AttributeSharingSessionDataStore.java

```diff
@@ -0,0 +1,141 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.security.session;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.eclipse.jetty.server.session.AbstractSessionDataStore;
+import org.eclipse.jetty.server.session.SessionContext;
+import org.eclipse.jetty.server.session.SessionData;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This implementation of {@link org.eclipse.jetty.server.session.SessionDataStore} communicates
+ * with the {@link AttributeSharingHashSessionIdManager} to retrieve the most up-to-date session
+ * attributes.
+ *
+ * <p>There is one <code>SessionDataStore</code> instance for each HTTP context on the Jetty server.
+ * In the current way we use pax-web with our jax-rs servers, session attributes are not shared
+ * across contexts. Our SSO solution necessitates a way to share those session attributes since the
+ * security token is stored on the session as an attribute.
+ */
+public class AttributeSharingSessionDataStore extends AbstractSessionDataStore {
+
+  private static final Logger LOGGER =
+      LoggerFactory.getLogger(AttributeSharingSessionDataStore.class);
+
+  private final Map<String, SessionData> sessionDataMap = new HashMap<>();
+  private AttributeSharingHashSessionIdManager attributeSharingHashSessionIdManager;
+
+  /**
+   * Called by the {@link AttributeSharingHashSessionIdManager} whenever it updates one of its
+   * session's attributes.
+   *
+   * @param id the session id
+   * @param sessionAttributes the session's attributes
+   */
+  public void updateSessionAttributes(String id, Map<String, Object> sessionAttributes) {
+    SessionData sessionData = sessionDataMap.get(id);
+    if (sessionData != null && !sessionData.getAllAttributes().equals(sessionAttributes)) {
+      LOGGER.debug(
```

Comment:
✏️ `trace` because this is normal program execution. See https://codice.atlassian.net/wiki/spaces/DDF/pages/7045131/Log+Levels

Reply:
I definitely remember changing this. I must've undone it somehow before committing 😅